modules/check_in/spec/services/v2/chip/service_spec.rb

```diff
@@ -337,4 +340,185 @@
+                          .demographic_confirmations).to eq(demographics_confirmation_hash)
+          end
+        end
+
+        context 'when no demographics data available in check_in_body' do
+          let(:demographics_confirmation_hash) do
+            {
+              demographicConfirmations: {}
+            }
+          end
+
+          it 'returns demographics confirmation hash with all demographics data' do
+            expect(subject.build(check_in: valid_check_in, params: {})
+                          .demographic_confirmations).to eq(demographics_confirmation_hash)
+          end
+        end
+      end
+
+      context 'with check_in_experience_no_demographics_confirmation_for_unverified_enabled turned off' do
+        before do
+          allow(Flipper).to receive(:enabled?)
+            .with(:check_in_experience_no_demographics_confirmation_for_unverified_enabled)
+            .and_return(false)
+        end
+
+        context 'when all demographics data available in check_in_body' do
+          let(:params) do
+            {
+              demographics_up_to_date: true,
+              next_of_kin_up_to_date: true,
+              emergency_contact_up_to_date: false
+            }
+          end
+
+          let(:demographics_confirmation_hash) do
+            {
+              demographicConfirmations: {
+                demographicsNeedsUpdate: false,
+                demographicsConfirmedAt: Time.zone.now.iso8601,
+                nextOfKinNeedsUpdate: false,
+                nextOfKinConfirmedAt: Time.zone.now.iso8601,
+                emergencyContactNeedsUpdate: true,
+                emergencyContactConfirmedAt: Time.zone.now.iso8601
+              }
+            }
+          end
+
+          it 'returns demographics confirmation hash with all demographics data' do
+            expect(subject.build(check_in: valid_check_in, params: params)
+                          .demographic_confirmations).to eq(demographics_confirmation_hash)
+          end
+        end
+
+        context 'when only demographics_up_to_date in check_in_body' do
+          let(:params) do
+            {
+              demographics_up_to_date: true
+            }
+          end
+
+          let(:demographics_confirmation_hash) do
+            {
+              demographicConfirmations: {
+                demographicsNeedsUpdate: false,
+                demographicsConfirmedAt: Time.zone.now.iso8601,
+                nextOfKinNeedsUpdate: true,
+                nextOfKinConfirmedAt: Time.zone.now.iso8601,
+                emergencyContactNeedsUpdate: true,
+                emergencyContactConfirmedAt: Time.zone.now.iso8601
+              }
+            }
+          end
+
+          it 'returns demographics confirmation hash with all demographics data' do
+            expect(subject.build(check_in: valid_check_in, params: params)
+                          .demographic_confirmations).to eq(demographics_confirmation_hash)
+          end
+        end
```

Comment:
The tests look good. I would suggest adding a test for when the flag is turned off and no demographic confirmation parameters are sent, as that is the most likely real life scenario.

Reply:
Basically, that's the issue getting fixed in this change.  Currently, if no demographic confirmation parameters are sent, the vets-api still sends `...NeedsUpdate` with true to the downstream.  

Highlighted test here covers the case where `next_of_kin_up_to_date` & `emergency_contact_up_to_date` not sent, but demographics_confirmation_hash sends a payload with `true` in `next_of_kin_up_to_date` & `emergency_contact_up_to_date`